engine-tests/src/test/java/org/terasology/config/flexible/SettingTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.terasology.config.flexible.validators.RangedNumberValidator;
+import org.terasology.engine.SimpleUri;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+import java.beans.PropertyChangeListener;
+
+import static org.junit.Assert.*;
+
+@RunWith(Enclosed.class)
+public class SettingTest {
+    public static class SetValue {
+        private Setting<Integer> setting;
+
+        private int eventResult;
+
+        @Before
+        public void setUp() {
+            setting = new SettingImpl<>(new SimpleUri("engine-tests:TestSetting"),
```

Comment:
Let's put the SimpleUri in a variable, to keep the constructor call cleaner visually.